acme-numbers-events.js

```diff
@@ -1,16 +1,88 @@
-console.log('hi');
+class Game {
+  constructor() {
+    this.gameNumbers = [];
+    this.sum = 0;
+    this.average = 0;
+  }
+  updateAverage() {
+    // let sum = this.gameNumbers.reduce(function(result, number) {
+    //   result += number;
+    //   return result;
+    // },0);
+    if (this.gameNumbers.length < 1) this.average = 0;
+    else this.average = Math.round(this.sum / this.gameNumbers.length);
+  }
+  newInstance() {
+    return new Game;
+  }
+}
 
-const button = document.getElementById('main-button');
+let currentGame = new Game;
+
+const mainButton = document.getElementById("main-button");
+const resetButton = document.getElementById("reset");
+const ulNumbers = document.getElementById("number-list");
+const averageElement = document.getElementById("average");
+
+ulNumbers.addEventListener('click', function(event) {
+  if(event.target.tagName === "LI") {
+    let number = event.target.innerText;
+    currentGame.sum -= number;
+    currentGame.gameNumbers.pop();
```

Comment:
would this always work? lets say you removed a number that's in the middle of the `gameNumbers` array, but you used `gameNumbers.pop()` in order to remove that number from the array
we actually removed the wrong number because we removed a number in the middle of the array, but we used `pop` to remove from the end! 

i realize that `gameNumbers` is basically being used just to store the amount of numbers- but you can achieve that same purpose with just a simple variable called `amountOfNumbers` :D 
once you add a new number, you do `amountOfNumbers++;`
once you remove a number, you do `amountOfNumbers--;`